### DISCUSSION TITLE
TSUN Smart App

### DISCUSSION THREAD

**Discussion by sberlin** [s-allius/tsun-gen3-proxy#195] [Q&A]
Hi, are the API endpoints for the new mobile app known? I reckon during the investigation for this project you stumbled over it at some point. I know of https://talent-monitoring.com/prod-api which is used on the old web and mobile apps. I cannot use them, because my refurbished MS-600 is still locked on another account. So I'm interested where to query for data of a GEN3 inverter when using only an account for TSUN Smart App.

I know of these projects already integrating the old API:
https://github.com/asciidisco/tsun-talent-monitoring
https://github.com/LenzGr/pytalent-monitor

**Comment by s-allius** [comment]
Hi sberlin,
This project uses a completely different approach. The tsun-gen3 proxy reads the data traffic from the inverter in the local network and you do not need an internet connection. The projects you mentioned get the data from the web API of the TSUN server.

So if you want to read and interpret the data from your inverter, this project is the right one for you. And the proxy is completely independent of the TSUN Cloud and of any TSUN account. If you want to use both the local MQTT of this proxy and the TSUN apps, you will of course need valid access data so that the proxy can forward the data to the TSUN Cloud. But this is optional.

**Reply by sberlin** [reply]
Yes, sure, I just thought you came across this information at some point. I'm looking for some raw data for further investigation of my setup without maintaining the stack, especially custom DNS. I think your project is impressive and genius to bypass the Cloud, though.

What I can do now, is to run the stack and check for domain queries when opening the app when connected to the same network. In addition packet sniffing on Android might be required.

**Reply by s-allius** [reply]
Properly your inverter will connect to `logger.talent-monitoring.com:5005`or `ìot.talent-monitoring.com:10000`
If you have a brand new one with SSL support he might connect to `iot.talent-monitoring.com:10443`.

If your inverter connects to one of the last two, you have a GEN3PLUS version and you can try `the client-mode`. Then the proxy connects to the port 8899 of the inverter und you must not touch your DNS setup

**Reply by sberlin** [reply]
My MS-600 is GEN3, so I'm stuck with custom DNS. But I might replace it eventually, because I have trouble with the startup voltage.

Thanks for sharing. I think in that case I will investigate on my own and we can close the discussion.
